main.js

```diff
@@ -276,25 +283,130 @@ function displayLargeCard() {
   renderAppState()
 }
 
-// function 7 - Define a function that takes a view name ('catalog' or 'details')
-// and adds a 'hidden' class to all data-view containers (see divs in html) that don't match that view.
 function addHiddenClass(viewName) {
   if (viewName === 'details') {
-    // adds hidden class to details app view (div)
-    $details.classList.add('hidden')
     $catalog.classList.remove('hidden')
+    $details.classList.add('hidden')
+    $cart.classList.remove('hidden')
   }
   if (viewName === 'catalog') {
-    // adds hidden class to catalog app view (div)
     $catalog.classList.add('hidden')
     $details.classList.remove('hidden')
+    $cart.classList.remove('hidden')
+  }
+  if (viewName === 'cart') {
+    $catalog.classList.remove('hidden')
+    $details.classList.remove('hidden')
+    $cart.classlist.add('hidden')
   }
 }
 
-// function 8 - Define a function that takes a cart object and returns a
-// DOM tree containing the count of items in the cart.
 function cartCount(cart) {
   var $cartCount = document.createElement('div')
-  $cartCount.textContent = 'Cart (' + cart.item.length + ')'
+  $cartCount.classList.add('cart-count')
+  $cartCount.textContent = 'Cart (' + cart.items.length + ')'
   return $cartCount
 }
+
+var $cartCounter = document.querySelector('.cart-count-container')
+$cartCounter.addEventListener('click', openTheCart)
+
+function openTheCart() {
+  $catalog.innerHTML = ''
+  $details.innerHTML = ''
+  app.view = 'cart'
+  renderAppState()
+}
+
+function createCartItem(item) { // change to (cartItem)?
```

Comment:
You should remove little comments like this.

Reply:
done
![screen shot 2018-07-05 at 1 13 51 pm](https://user-images.githubusercontent.com/13921070/42345787-590882b6-8055-11e8-893a-006881c87065.png)